src/components/navigation/components/LoginController.js

```diff
@@ -184,49 +202,73 @@ export class LoginController extends React.PureComponent {
                     // Alerts.deQueue(Alerts.LoggedOut);
                     setUserID(null);
 
-                    // If is programatically called with error CB, let error CB handle everything.
-                    var errorCallbackFxn = typeof errorCallback === 'function' ? errorCallback : this.loginErrorCallback;
-                    errorCallbackFxn(error);
+                    if (typeof errorCallback === "function") {
+                        errorCallback(error);
+                    }
                 });
 
         });
+    }
 
+    auth0LoginCallback(authResult){
+        // First stage: we just have gotten JWT from the Auth0 widget but have not auth'd it against it our own system
+        // to see if this is a valid user account or some random person who just logged into their Google account.
```

Comment:
While I do like inline comments like these, they aren't a replacement for docstrings IMO. The inline comments aren't always helpful if you don't understand how what it's doing fits into the overall scheme. This is more of a general comment as there are things you're doing to mitigate this ie: good function naming.